### ISSUE TITLE
Direction of the trajectory

### ISSUE THREAD

**Issue by huang-1030** [orangeduck/Motion-Matching#31]
When I don't want to use the positive direction of the hip node as the direction of the trajectory, how should I change the Trajectory Directions? When I try to use the predicted three position information, the resulting bin file will directly change to nan when training the decompressor.

**Comment by orangeduck** [comment]
Sorry I'm not sure what you want to do.

**Comment by huang-1030** [comment]
Sorry, let me rephrase. In your example, the action animations were some body movements that were forward facing, but now some of my action animations are not facing forward, as shown in my picture. What should I do to obtain the correct orientation when I use generate_database.py to generate a new bin again. I tried to change the result of line 123,124 in generate_database.py, and also tried to use the predicted positions of frames 20, 40, and 60 as future directions. However, after training, when loaded into your C++project,I didn't get want results. During the controller's walking process, move the body forward from the front instead of the side as shown in the picture. How should I handle this
![1717638734872](https://github.com/orangeduck/Motion-Matching/assets/84122922/318b408a-c6d8-40ae-933c-afb9dfa74ccf)

**Comment by huang-1030** [comment]
> 抱歉，我不确定你想做什么。
Sorry, let me rephrase. In your example, the action animations were some body movements that were forward facing, but now some of my action animations are not facing forward, as shown in my picture. What should I do to obtain the correct orientation when I use generate_database.py to generate a new bin again. I tried to change the result of line 123,124 in generate_database.py, and also tried to use the predicted positions of frames 20, 40, and 60 as future directions. However, after training, when loaded into your C++project,I didn't get want results. During the controller's walking process, move the body forward from the front instead of the side as shown in the picture. How should I handle this
![Uploading 1717638734872.png…]()

**Comment by orangeduck** [comment]
I think you can use the direction of travel as the facing direction however it wont be as simple as just taking the vector towards the future root position and normalizing it - this will be very unstable for cases like when the character is standing still. You will have to come up with your own script for extracting it procedurally from the animation data and make sure it is robust to all sorts of situations.

**Comment by huang-1030** [comment]
> I think you can use the direction of travel as the facing direction however it wont be as simple as just taking the vector towards the future root position and normalizing it - this will be very unstable for cases like when the character is standing still. You will have to come up with your own script for extracting it procedurally from the animation data and make sure it is robust to all sorts of situations.

Okay, I'll try again. Thank you for your suggestion.
